bluesky_widgets/qt/run_tree.py

```diff
@@ -0,0 +1,195 @@
+from qtpy import QtCore
+from qtpy.QtCore import (
+    QAbstractItemModel,
+    QModelIndex,
+    QTimer,
+    Qt,
+)
+from qtpy.QtWidgets import QAbstractItemView, QHeaderView, QTreeView
+
+from databroker import catalog
+from databroker.core import BlueskyEventStream
+
+class RunTree(object):
+    """Lazily populate the tree as data is requested. """
+    def __init__(self, run):
```

Comment:
A long-term comment, not implying the need for any changes here (for the present):

I am intrigued by applying the pattern `QtThing(model)` everywhere in this library, as we started to do with `QtSearch(seach_model: SearchModel)` and so on with its sub-components. In this case, I think the `model` would signal things like "just received a new {stream, descriptor, event, stop document}" if the `BlueskyRun` instance in question were a live one, and the tree would update itself accordingly.

It seems at least _plausible_ that the `model` in question here could just be a `BlueskyRun` with observable signals, provided by subclassing or by injection. I have some yet-to-be-finished-or-pushed code in this direction that I think will match up with this well.

However, if we want to be able to "hot swap" which Run we are viewing in this Tree, then maybe we want the extra layer of indirection of a `model` which has-a BlueskyRun, and can have different BlueskyRuns assigned to it during its lifecycle.

Reply:
It would be great to explore, I think you still need the underlying tree model with some knowledge of BlueskyRun, but it would be nice to build more intelligence into how we might update this. Maybe we can improve this in further feature branches, I think I may need a little more help seeing how best to apply what you mention here.